### PR TITLE
Reformat event numbers as BN

### PR DESCRIPTION
**Pull request by cgewecke**
#1165 

Companion to #1145.  

+ Renamed the reformatter method so it's more general.
+ Added test for numerical event outputs

### CODE REVIEW

File: packages/truffle-contract/lib/reformat.js [hunk 1]
```diff
@@ -37,14 +37,14 @@ const _convertNumberArray = function(arr, format){
  *   - object (with index keys and optionally, named keys)
  *   - array
  *   - single primitive
- * @param  {String|Object|Array} result  web3 call result
- * @param  {Array}               outputs ABI outputs array
+ * @param  {String|Object|Array} result      web3 call result
+ * @param  {Array}               abiSegment  event inputs OR .call outputs
```
Comment (gnidan) [inline packages/truffle-contract/lib/reformat.js]: I think these are called `parameters` in the ABI spec

Comment (cgewecke) [reply packages/truffle-contract/lib/reformat.js]: Why are they so normie over there?

Comment (gnidan) [reply packages/truffle-contract/lib/reformat.js]: Ugh, normies!

**Comment by cgewecke** [comment]
@gnidan Ah! Ok. Well now it's just total muddle.

**Pull request merged**